src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -3440,67 +3626,116 @@ class CgroupUtils:
+            PBS_MOM_HOME = pbs_conf['PBS_MOM_HOME']
+    # Try reading the config file
+    if not PBS_EXEC or not PBS_HOME or not PBS_MOM_HOME:
+        if 'PBS_CONF_FILE' in os.environ:
+            pbs_conf_file = os.environ['PBS_CONF_FILE']
         else:
-            pbs.logmsg(pbs.EVENT_DEBUG,
-                       "PBS_EXEC needs to be defined in the config file")
-            pbs.logmsg(pbs.EVENT_DEBUG, "Exiting the cgroups hook")
-            pbs.event().accept()
+            pbs_conf_file = os.path.join(os.sep, 'etc', 'pbs.conf')
+        regex = re.compile(r'\s*([^\s]+)\s*=\s*([^\s]+)\s*')
+        try:
+            with open(pbs_conf_file, 'r') as desc:
+                for line in desc:
+                    match = regex.match(line)
+                    if match:
+                        if not PBS_EXEC and match.group(1) == 'PBS_EXEC':
+                            PBS_EXEC = match.group(2)
+                        if not PBS_HOME and match.group(1) == 'PBS_HOME':
+                            PBS_HOME = match.group(2)
+                        if not PBS_MOM_HOME and (match.group(1) ==
+                                                 'PBS_MOM_HOME'):
+                            PBS_MOM_HOME = match.group(2)
+        except:
+            pass
+    # If PBS_MOM_HOME is not set, use the PBS_HOME value
+    if not PBS_MOM_HOME:
+        PBS_MOM_HOME = PBS_HOME
+    # Sanity check to make sure each global path is set
+    if not PBS_EXEC:
+        pbs.logmsg(pbs.EVENT_DEBUG, "Unable to determine PBS_EXEC. Exiting.")
+        event.accept()
+    if not PBS_HOME:
+        pbs.logmsg(pbs.EVENT_DEBUG, "Unable to determine PBS_HOME. Exiting.")
+        event.accept()
+    if not PBS_MOM_HOME:
+        pbs.logmsg(pbs.EVENT_DEBUG,
+                   "Unable to determine PBS_MOM_HOME. Exiting.")
+        event.accept()
     # Determine PBS Pro version
-    cmd = os.path.join(pbs_exec, 'bin', 'pbs_hostn')
+    cmd = os.path.join(PBS_EXEC, 'bin', 'pbs_hostn')
     process = subprocess.Popen([cmd, '--version'], shell=False,
                                stdout=subprocess.PIPE)
     stdout = process.communicate()[0].splitlines()
-    regex = re.compile('\D*(\d+).(\d+).(\d+).*')
-    m = regex.match(stdout[0])
-    if m is None:
+    regex = re.compile(r'\D*(\d+).(\d+).(\d+).*')
+    match = regex.match(stdout[0])
+    if match is None:
         pbs.logmsg(pbs.EVENT_DEBUG, "Failed to detemine PBS Pro version")
         pbs.logmsg(pbs.EVENT_DEBUG, "Exiting the cgroups hook")
-        pbs.event().accept()
-    pbs_version = (int(m.group(1)), int(m.group(2)), int(m.group(3)))
+        event.accept()
+    PBS_VERSION = (int(match.group(1)),
+                   int(match.group(2)),
+                   int(match.group(3)))
     pbs.logmsg(pbs.EVENT_DEBUG4, "%s: PBS Pro version is %d.%d.%d" %
-               (caller_name(), pbs_version[0], pbs_version[1], pbs_version[2]))
+               (caller_name(), PBS_VERSION[0], PBS_VERSION[1], PBS_VERSION[2]))
```

Comment:
I hate to remove it in case we do in the future.

Reply:
This will end up in the mom logs.  The mom will print its version when it starts up.  Printing the version of PBS as the hook found it is not really helpful.  I really don't see the need.

Reply:
Useful for debugging when looking at a log someone has sent us.

Reply:
It is in the logs.  When mom starts she prints her version in the logs.

Reply:
This was a new addition to 18 that was added a few months back.

Reply:
I still don't think it is necessary.  This goes into the mom logs.  Mom prints her version in the logs.  This is the same version.  I see no reason to print this once per hook invocation.

Reply:
I was not saying that this is still needed. I was just providing info.

Reply:
Removed.